aspnetcore/host-and-deploy/iis/troubleshoot.md

```diff
@@ -15,46 +15,140 @@ uid: host-and-deploy/iis/troubleshoot
+**500 Internal Server Error**  
+The app starts, but an error prevents the server from fulfilling the request.
 
-In most cases, enabling application logging assists in troubleshooting problems with the app or the reverse proxy. See [Logging](xref:fundamentals/logging/index) for more information.
+This error occurs within the app's code during startup or while creating a response. The response may contain no content, or the response may appear as a *500 Internal Server Error* in the browser. The Application Event Log usually states that the app started normally. From the server's perspective, that's correct. The app did start, but it can't generate a valid response. [Run the app at a command prompt](#run-the-app-at-a-command-prompt) on the server or [enable the ASP.NET Core Module stdout log](#aspnet-core-module-stdout-log) to troubleshoot the problem.
 
-The last troubleshooting tip pertains to apps that fail to run after upgrading either the .NET Core SDK on the development machine or package versions within the app. In some cases, incoherent packages may break an app when performing major upgrades. Most of these issues can be fixed by:
+## Default startup limits
 
-* Deleting the `bin` and `obj` folders in the project.
-* Clearing package caches at `%UserProfile%\.nuget\packages\` and `%LocalAppData%\Nuget\v3-cache`.
-* Restoring and rebuilding the project.
-* Confirming that the prior deployment on the server has been completely deleted prior to re-deploying the app.
+The ASP.NET Core Module is configured with a default *startupTimeLimit* of 120 seconds. When left at the default value, an app may take up to two minutes to start before the module logs a process failure. For information on configuring the module, see [Attributes of the aspNetCore element](xref:host-and-deploy/aspnet-core-module#attributes-of-the-aspnetcore-element).
+
+## Troubleshoot app startup errors
+
+### Application Event Log
+
+Access the Application Event Log:
+
+1. Open the Start menu, search for **Event Viewer**, and then select the **Event Viewer** app.
+1. In **Event Viewer**, open the **Windows Logs** node.
+1. Select **Application** to open the Application Event Log. Search for errors associated with the failing app.
+
+### Run the app at a command prompt
+
+Many startup errors don't produce useful information in the Application Event Log. You can find the cause of some errors by running the app at a command prompt on the hosting system.
+
+**Framework-dependent deployment**
+
+If the app is a [framework-dependent deployment](/dotnet/core/deploying/#framework-dependent-deployments-fdd):
+
+1. At a command prompt, set the `ASPNETCORE_DETAILEDERRORS` environment variable. Use the command: `set ASPNETCORE_DETAILEDERRORS=true`.
```

Comment:
ASPNETCORE_DETAILEDERRORS doesn't really apply when running from the command line. It's more applicable to the IIS/Express scenario as they enable the Capture Startup Errrors option.